### PR TITLE
- Implementation of DNSSEC02 updated

### PR DESCRIPTION
**Pull request by vlevigneron**
- Implementation is now conformant with the last spec of this test case.
- Fixes zonemaster/zonemaster-engine#619

### CODE REVIEW

**Comment by vlevigneron** [comment]
**TRANSLATION** not yet done

The following MSGID has been added:

```
              BROKEN_DS
              BROKEN_RRSIG
              DNSKEY_KSK_NOT_SEP
              DNSKEY_NOT_ZONE_SIGN
              DS_MATCHES
              NO_MATCHING_DNSKEY
              NO_MATCHING_RRSIG
              NO_RRSIG_DNSKEY
```
The following MSGID has been deleted:

```
              NO_DS
              DS_FOUND
              DS_RFC4509_NOT_VALID
              COMMON_KEYTAGS
              NO_COMMON_KEYTAGS
              DS_MATCHES_DNSKEY
              DS_DOES_NOT_MATCH_DNSKEY
              DS_MATCH_FOUND
              DS_MATCH_NOT_FOUND
```

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 1]
```diff
@@ -511,9 +512,17 @@ Readonly my %TAG_DESCRIPTIONS => (
...
           @_;
     },
-    COMMON_KEYTAGS => sub {
-        __x    # DNSSEC:COMMON_KEYTAGS
-          'There are both DS and DNSKEY records with key tags {keytags}.', @_;
+    BROKEN_DS => sub {
+        __x    # DNSSEC:BROKEN_DS
+          'DNSKEY record with tag {keytag} returned by nameserver {ns}/{address} doe not match '
+          . 'DS record with same tag in parent zone.',
+          @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:519]: 'DNSKEY record with tag {keytag} returned by nameserver {ns}/{address} does not match the algorithm and hash
values in a DS record with same tag in parent zone.'

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 2]
```diff
@@ -610,32 +631,9 @@ Readonly my %TAG_DESCRIPTIONS => (
...
-          'No DS record with a matching DNSKEY record was found.', @_;
-    },
-    DS_RFC4509_NOT_VALID => sub {
-        __x    # DNSSEC:DS_RFC4509_NOT_VALID
-          'Existing DS with digest type 2, while they do not match DNSKEY records, '
-          . 'prevent use of DS with digest type 1 (RFC4509, section 3).',
-          @_;
+    DS_MATCHES => sub {
+        __x    # DNSSEC:DS_MATCHES
+          'Found DS records matching DNSKEY records.',  @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:636]: 'The DS records in the parent zone match DNSKEY records in the child zone.'

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 3]
```diff
@@ -738,6 +728,18 @@ Readonly my %TAG_DESCRIPTIONS => (
           . '{sigs} RRSIG records and {soas} SOA records.',
           @_;
     },
+    NO_MATCHING_DNSKEY => sub {
+        __x    # DNSSEC:NO_MATCHING_DNSKEY
+          'Although DS record is present in parent, nameserver {ns}/{address} returned no DNSKEY '
+          . 'record with same tag {keytag}.',
+          @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:735]: 'Nameserver {ns}/{address} returned no DNSKEY record matching the DS record with tag {keytag} found in the parent zone.'

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 4]
```diff
@@ -738,6 +728,18 @@ Readonly my %TAG_DESCRIPTIONS => (
...
+    },
+    NO_MATCHING_RRSIG => sub {
+        __x    # DNSSEC:NO_MATCHING_RRSIG
+          'Although DS record is present in parent, nameserver {ns}/{address} returned no signature '
+          . 'corresponding to the DNSKEY with tag {keytag}.',
+          @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:741]: 'Nameserver {ns}/{address} returned no signature on the DNSKEY RRset that corresponds to the DNSKEY with tag {keytag} even though there is a DS record in the parent zone for that DNSKEY record.'

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 5]
```diff
@@ -511,9 +512,17 @@ Readonly my %TAG_DESCRIPTIONS => (
...
+        __x    # DNSSEC:BROKEN_DS
+          'DNSKEY record with tag {keytag} returned by nameserver {ns}/{address} doe not match '
+          . 'DS record with same tag in parent zone.',
+          @_;
+    },
+    BROKEN_RRSIG => sub {
+        __x    # DNSSEC:BROKEN_RRSIG
+          'Signature for DNSKEY with tag {keytag} returned by nameserver {ns}/{address} failed'
+          . 'to verify with error \'{error}\' although DS with same tag is present in parent.',
+          @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:525]: 'The RRSIG of the DNSKEY RRset created by tag {keytag} returned by nameserver {ns}/{address} failed to be verified with error \'{error}\' (a DS redord with same tag is present in the parent zone).'

          'Signature for DNSKEY with tag {keytag} returned by nameserver {ns}/{address} failed'
          . 'to verify with error \'{error}\' although DS with same tag is present in parent.'

**Review by matsduf — CHANGES_REQUESTED** [review]

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 6]
```diff
@@ -610,32 +632,9 @@ Readonly my %TAG_DESCRIPTIONS => (
...
-          'No DS record with a matching DNSKEY record was found.', @_;
-    },
-    DS_RFC4509_NOT_VALID => sub {
-        __x    # DNSSEC:DS_RFC4509_NOT_VALID
-          'Existing DS with digest type 2, while they do not match DNSKEY records, '
-          . 'prevent use of DS with digest type 1 (RFC4509, section 3).',
-          @_;
+    DS_MATCHES => sub {
+        __x    # DNSSEC:DS_MATCHES
+          'The DS records in the parent zone match DNSKEY records in the child zone..',  @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:637]: One dot too many, "zone.."

**Review by matsduf — APPROVED** [review]

**Pull request merged**